lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -56,7 +56,7 @@ def non_threadsafe_bulk(actions)
 
         args.delete(:_upsert)
 
-        if source
+        if source and action != 'delete'
```

Comment:
Can you change `and` to `&&`. We don't use `and` on the logstash project because weak precedence causes bugs in weird ways :(.

Once this is in I'll be glad to merge. Thank you so much for contributing, and sorry I missed this earlier today.